app/src/test/java/com/nervousfish/nervousfish/data_objects/SimpleKeyTest.java

```diff
@@ -98,4 +116,37 @@ public void testHashCodeNotNull() {
         assertNotNull(key.hashCode());
     }
 
+    @Test
+    public void testToJson() throws IOException {
+        IKey key = new SimpleKey("foo", "bar");
+        JsonWriter writer = mock(JsonWriter.class);
+        when(writer.name(anyString())).thenReturn(writer);
+        key.toJson(writer);
+        verify(writer).name("name");
+        verify(writer).name("key");
+        verify(writer).value("foo");
+        verify(writer).value("bar");
+    }
+
+    @Test
+    public void testSerialization() {
+        final IKey key = new SimpleKey("foo", "bar");
+        try (
+                ByteArrayOutputStream bos = new ByteArrayOutputStream();
+                ObjectOutputStream oos = new ObjectOutputStream(bos)
+        ) {
+            oos.writeObject(key);
+            byte[] bytes = bos.toByteArray();
+            try (ByteArrayInputStream bis = new ByteArrayInputStream(bytes);
+                 ObjectInputStream ois = new ObjectInputStream(bis)) {
+                IKey key1 = (IKey) ois.readObject();
+                assertTrue(key1.getName().equals("foo"));
+                assertTrue(key1.getKey().equals("bar"));
+            } catch (IOException | ClassNotFoundException e) {
+                e.printStackTrace();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
I think this way the test doesn't fail when the exception is throw? Should you just add a `throws IOException` to the method definition? (Also applies to similar tests in other test classes)